File: goversion/mycd.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func sub(a []string, c string) bool {
	var sts bool = false
	for _, k := range a {
		if c == k {
			fmt.Println(k, c)
			sts = true
			break
		}
	}
	return sts
}

func main() {
	var cur string
	var new string
	var joined string
	//var prev string
	if len(os.Args) <= 1 {
		fmt.Println("[ARGUMENT ERROR] mycd takes only two input paths.")
		os.Exit(1)
	}
	if len(os.Args) > 3 {
		fmt.Println("[ARGUMENT ERROR] mycd takes only two input paths.")
		os.Exit(1)
	}
	cur = os.Args[1]
	new = os.Args[2]

	curs := strings.Split(cur, "/")
	news := strings.Split(new, "/")
	temp := curs
	//err := false
	//curl := len(news)

	for _, m := range news {
		if !sub(curs, m) && m != ".." && m != "" {
			temp = append(temp, m)
		} else if !sub(curs, m) && m == ".." && m != "" {
			temp = temp[:len(temp)-1]
		}
	}

	for _, k := range temp {
		joined += k + "/"
	}

	if len(joined) == 0 {
		joined = "/"
	}

	fmt.Println(joined)
}

File: cpversion/mycd.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<string> split(string st,char s){
    int a=0;
    vector<string> out;
    string temp;
    for(int i=0;i<st.length();i++){
        if(s[i]==s || i==st.length()){
            out.insert(out.end(),temp);
            temp="";
        }else{
            temp+=st[i];
        }
    }

    return out;
}

bool isin(vector<string> b,string s){
    bool sts = 0;
    for(auto i=b.begin();i<b.end();i++){
        if(*i==s){
            sts=1;
            break;
        }
    }
    return sts;
}

bool sub(string s,string sub){
    string temp;
    bool sts = 0;
    for(int i=0;i<s.length();i+=sub.length()){
        for(int k=i;k<(i+sun.length());k++){
            temp+=s[k];
            if(temp == sub){
                sts = 1;
                break;
            }
        }
        if(sts){
            break;
        }
    }
    return sts;
}

string dirCheck(vector<string> p,vector<string> c){
    string out;
    vector<string> temp;
    bool err = 0;
    temp = p;
    for(auto i=c.begin();i<c.end();i++){
        if(sub(*i,"..")&& *i != ".."){
            cout << "ERROR: Invalid dir name" << endl;
            err = 1;
            break;
        }
        if(*i=="") continue;
        if(!isin(p,*i) && *i != ".."){
            temp.insert(te,p.end(),*i);
        }else if(!isin(p,*i) && *i == ".."){
            temp.erase(temp.end());
        }
    }

    if(err){
        return "";
    }

    for(auto i=temp.begin();i<temp.end();i++){
        out += *i + "/";
    }

    return out;
}

int main(int argc,char **argv){
    vector<string> cur = split(argv[1].'/');
    vector<string> news = split(argv[2].'/');
    cout << dirCheck(cur,news) << endl;
}